src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -358,6 +375,20 @@ func (d *downsamplerAndWriter) writeToStorage(
 	return multiErr.FinalError()
 }
 
+func (d *downsamplerAndWriter) maybeRecordDatapointsWithEmptyAnnotation(q *storage.WriteQuery) {
+	if len(q.Annotation()) == 0 {
+		n := int64(len(q.Datapoints()))
+		switch q.Attributes().MetricsType {
+		case storagemetadata.UnaggregatedMetricsType:
+			d.metrics.unaggregatedDatapointsWithEmptyAnnotation.Inc(n)
+		case storagemetadata.AggregatedMetricsType:
+			d.metrics.aggregatedDatapointsWithEmptyAnnotation.Inc(n)
+		default:
+			// ignore
```

Comment:
There are other types, though I'm not sure if those should be taken into account

https://github.com/m3db/m3/blob/d95a5dfa41ec02af603bfce3e421ac473d11dce8/src/query/storage/m3/storagemetadata/types.go#L32-L40

Reply:
Looks like `storageAttributesFromPolicy` can only return either `UnaggregatedMetricsType` or `AggregatedMetricsType`. If you want to be fully covered, you could add some kind of "storage-policy: other" label for the `default` branch of the `switch` statement (but I'm not advocating for this).